google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -56,7 +56,13 @@ void ListObjects(google::cloud::storage::Client client, int& argc,
   //! [list objects] [START storage_list_files]
   namespace gcs = google::cloud::storage;
   [](gcs::Client client, std::string bucket_name) {
-    for (gcs::ObjectMetadata const& meta : client.ListObjects(bucket_name)) {
+    for (auto&& m : client.ListObjects(bucket_name)) {
```

Comment:
I think you may want to call ListObjects with `StatusOr<T>` then get the objects in the bucket. Err, though I think this example of StatusOr is verifying the underlying GET's for the next set of objects didn't fail. 

At first glance it's a bit confusing because I get a StatusOr for each ObjectMetadata and I would think a request occurs per object.

Reply:
Apologies, what I meant was the sample is confusing as it iterates over a list of objects it checks the status of the object.

Though it is confusing, I can't think of an alternative. I hope that makes sense.